app/models/service_container_template.rb

```diff
@@ -0,0 +1,104 @@
+class ServiceContainerTemplate < ServiceGeneric
+  delegate :container_template, :container_manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, add_options = {})
+    unless add_options.blank?
+      _log.info("Override with new options:")
+      $log.log_hashes(add_options)
+    end
+
+    save_job_options(action, add_options)
+  end
+
+  def execute(action)
+    opts = get_job_options(action)
+
+    _log.info("Container template provisioning with options:")
+    $log.log_hashes(opts)
+
+    params = process_parameters(opts[:parameters])
+    ManageIQ::Providers::Openshift::ContainerManager::OrchestrationStack.create_stack(container_template, params, opts[:container_project_name])
+    _log.info("Provisioning of container template with ID: [#{id}] name:[#{name}] was initiated.")
+  end
+
+  def check_completed(action)
+    done = stack(action).provider_objects.all? { |obj| obj.resource_category.constantize.find_by(:name => obj.name) }
+    message = done ? "completed" : "not completed yet"
```

Comment:
Do we have another message string we use besides "not completed yet"?  I'm thinking "in progress" but we should be consistent here if possible.